pkg/health/checker/checker.go

```diff
@@ -57,32 +60,17 @@ func (c consulHealthChecker) WatchNodeService(
 ) {
 	defer close(resultCh)
 
-	var curIndex uint64 = 0
-
 	for {
 		select {
 		case <-quitCh:
 			return
 		case <-time.After(1 * time.Second):
-			var err error
-			var checks []*api.HealthCheck
-			checks, curIndex, err = c.fetchNodeHealth(nodename, curIndex)
+			kvCheck, err := c.consulStore.GetHealth(serviceID, nodename)
 			if err != nil {
 				errCh <- err
+			} else {
+				resultCh <- consulWatchToResult(kvCheck)
```

Comment:
blergh polling... luckily only pkg/replication uses this function, and technically it's not _worse_ than it was before, so we can punt on improving this.